cmd/clusterctl/cmd/describe_cluster_test.go

```diff
@@ -331,3 +335,31 @@ func withDeletionTimestamp(object ctrlclient.Object) {
 	now := metav1.Now()
 	object.SetDeletionTimestamp(&now)
 }
+
+type Table struct {
+	tableData []string
+}
+
+func MatchTable(expected []string) gtype.GomegaMatcher {
+	return &Table{tableData: expected}
+}
+
+func (t *Table) Match(actual interface{}) (bool, error) {
+	tableString := actual.(string)
+	tableParts := strings.Split(tableString, "\n")
+
+	for i := range t.tableData {
+		if !strings.HasPrefix(tableParts[i], t.tableData[i]) {
+			return false, fmt.Errorf("The Table does not match expected table output")
```

Comment:
I dont think you should return an error here instead just return false if the table data does not match.
Example if the table data does not match the test output looks like this:
```
--- FAIL: Test_TreePrefix (0.00s)
    --- FAIL: Test_TreePrefix/Second_level_child_should_get_the_right_prefix (0.00s)
        /home/vmware/workspace/go/src/sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/describe_cluster_test.go:292: 
            The Table does not match expected table output
FAIL
FAIL	sigs.k8s.io/cluster-api/cmd/clusterctl/cmd	0.062s
FAIL
```

Notice that it doest show any information about actual and expected values.